lib/services/iotaRedirector.js

```diff
@@ -310,27 +311,13 @@ function processRequests(req, res, next) {
                             ']'
                     )
                 );
-            } else if (body && body.length > 0) {
-                let parsedBody;
+            } else if (body) {
                 let parseError;
-
-                try {
-                    parsedBody = JSON.parse(body);
-                    if (response.statusCode === 409) {
-                        parseError = new errors.DuplicateError(parsedBody);
-                    }
-                    logger.debug(context, 'Response body from the redirected request parsed: \n\n%j\n', parsedBody);
-                } catch (e) {
-                    logger.error(
-                        context,
-                        'REDIRECTION-003: Error parsing response body from the redirected request: %s to [%s]',
```

Comment:
Note this is an error document at https://github.com/telefonicaid/iotagent-manager/blob/master/docs/operations.md#redirection-003-error-parsing-response-body-from-the-redirected-request-s

I understand that if we merge the code as it is now we lose the the ability of capturing and logging such error, so removing part of the current IOTA Manager functionality. That should be avoided.

If I understand correctly from previous comments now is `got` the one that does payload parsing and detect JSON parsing problem so in that case the logic goes in the block at L296-L302 (please tell me if I'm wrong). 

```
            if (error) {
                logger.error(
                    context,
                    'REDIRECTION-001: Error found redirecting requests to [%s]: %j',
                    options.uri,
                    error
                );
                callback(new errors.TargetServerError(error));
            }
```

I understand the processing of `error` should be more fine-grained so it can distinguish between REDIRECTION-001 and REDIRECTION-003 and log it properly, as we are currently doing with `requests` module.

Reply:
Fixed b10dd51 - added a test to give a garbage non-JSON response and re-added the code to log it.